substrate/frame/proxy/src/lib.rs

```diff
@@ -392,7 +397,7 @@ pub mod pallet {
 			let announcement = Announcement {
 				real: real.clone(),
 				call_hash,
-				height: system::Pallet::<T>::block_number(),
+				height: T::BlockNumberProvider::current_block_number(),
```

Comment:
The migration would need to convert all old block numbers into new ones. We could provide a migration for teams upgrading these pallets that assumes the old times were 12 seconds and the new ones 6 seconds since it's the relay chain?

Reply:
Couldn't you just define three `const`s in the migration type (e.g. [here](https://github.com/paritytech/polkadot-sdk/blob/e542796/substrate/frame/identity/src/migration.rs#L70))? They would be `old_block_time` (12s), `new_block_time` (6s), and `offset` (to account for the different block heights).

Reply:
https://github.com/paritytech/polkadot-sdk/pull/5656#issuecomment-2352671254
^^ This would be a better strategy than migration.